lib/IDE/CodeCompletion.cpp

```diff
@@ -3697,8 +3700,8 @@ class CompletionLookup final : public swift::VisibleDeclConsumer {
       }
 
       // Otherwise, check the result type matches the contextual type.
-      auto declTy = getTypeOfMember(VD, T);
-      if (declTy->hasError())
+      auto declTy = T->getTypeOfMember(CurrModule, VD);
```

Comment:
Are the changes below to `mapTypeIntoContext` supposed to rectify this?  If so, how does that compare to `eraseArchetypes`?

Reply:
We want to keep `ArchetypeType` here because `T` might be an 'ArchetypeType' and we want to check if result of `declTy` is convertible to `T`.

Let's say `T` is archetype `U` which conforms to `HasStatic` protocol and `VD` is `static var instance: Self { get }`, `getTypeOfMember(VD, T)` returns `HasStatic` protocol, but that is not convertible to archetype `U`. But `T->getTypeOfMember(CurrModule, VD)` correctly returns `U`.

This is just a filtering logic that checks `VD` can be referenced by 'implicit member expression' syntax in `T` type context. The result of `declTy` isn't used anywhere else. The type name for completion result is re-constructed by result builder.

Reply:
Okay, thanks for explaining!